src/Microsoft.Framework.Runtime/NuGet/Utility/PathValidator.cs

```diff
@@ -78,7 +81,18 @@ public static bool IsValidUrl(string url)
             Uri result;
 
             // Make sure url starts with protocol:// because Uri.TryCreate() returns true for local and UNC paths even if badly formed.
-            return Regex.IsMatch(url, @"^\w+://", RegexOptions.IgnoreCase) && Uri.TryCreate(url, UriKind.Absolute, out result);
+            var urlParts = url.Split(new string[] { "://" }, 2, StringSplitOptions.None);
+            if (urlParts == null || urlParts.Length != 2)
+            {
+                return false;
+            }
+
+            if (urlParts[0].Any(c => !char.IsLetterOrDigit(c) && c != '_'))
```

Comment:
nah..contexts are bit different. One is url validation, other package name validation etc.